### PR TITLE
🚀Added Gmplot

### LINKED ISSUE #713: 🌍GPS - Track The Route
# Description 
- I have been provided with a [`CSV`](https://www.bigcommerce.com/ecommerce-answers/what-csv-file-and-what-does-it-mean-my-ecommerce-business/#:~:text=A%20CSV%20is%20a%20comma,Microsoft%20Excel%20or%20Google%20Spreadsheets.)(comma separated value) file in which the **latitudes** and **longitudes** of a specific area are written
_So_ we can track the whole route of a person by reading values of that file on Google Maps and create an Image containing route or location .
_OR_ Just point the location on the map .

# Details 
- API Used [`gmplot`](https://github.com/gmplot/gmplot/wiki)

# What to do
- Take CSV file input **or** any latitudes or longitudes-> plot the map -> give output in image format

### PR DESCRIPTION
**Pull request by Lakhankumawat**
# Description
 Tracking the route of a person _OR_ Tracking a specific path followed by a person 
- Gmplot Can be used to plot the coordinates on Google map .
- These coordinates can be imported from a fixed CSV file that doesn't change  _OR_ can be taken as input from user in Latitude and Longitude format .

## Have you read the [Contributing Guidelines on Pull Requests](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/CONTRIBUTING.md)?

- [x] Yes
- [ ] No

## Type of change

- [x] New feature (non-breaking change which adds functionality)
 
## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/Lakhankumawat/Amazing-Python-Scripts/blob/gmplot/Gmplot-Track%20the%20Route/README.md)
- [x] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by Lakhankumawat** [comment]
For any  modifications let me know .

File: Gmplot-Track the Route/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    37.774797, -122.454538
+    37.771988, -122.454018
+    37.773646, -122.440979
+    37.772742, -122.440797
+    37.771096, -122.453889
+    37.768669, -122.453518
+    37.766227, -122.460213
+    37.764028, -122.510347
+
+ # Author
```
Comment (kaustubhgupta) [inline Gmplot-Track the Route/README.md:35]:
```suggestion
# Author 

```

File: Gmplot-Track the Route/main.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+import csv  #csv - Comma Separated Values
+from gmplot import gmplot
+
+
+gmap = gmplot.GoogleMapPlotter(20.613456, 72.9431185,17) #Plotting data on map
+#gmap.coloricon="https://developers.google.com/maps/documentation/javascript/examples/full/images/beachflag.png"
+
+with open('Home/MyPC/Python/LatLong.csv','r') as f:  #give the address of csv file with respect to your folder containing .py file
```
Comment (kaustubhgupta) [inline Gmplot-Track the Route/main.py:9]: ask the user to input the file path

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
upload the CSV used in the script. Implement these changes too:

File: Gmplot-Track the Route/README.md (hunk 1 above)
Comment (Lakhankumawat) [reply Gmplot-Track the Route/README.md:35]: ### I got to changes which are requested but i have some doubts 
1. Do  i remove author name from README / I mean i didn't get above review.
2. Rest else is fine i will take address as an input form user .

Comment (kaustubhgupta) [reply Gmplot-Track the Route/README.md:35]: No, You can keep the author name, no issues, I was suggesting to remove the space with (#) and line start as the Author level was not rendered properly.

File: Gmplot-Track the Route/main.py (hunk 2 above)
Comment (Lakhankumawat) [reply Gmplot-Track the Route/main.py]: ### Requested changes are made .

File: Gmplot-Track the Route/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    37.774797, -122.454538
+    37.771988, -122.454018
+    37.773646, -122.440979
+    37.772742, -122.440797
+    37.771096, -122.453889
+    37.768669, -122.453518
+    37.766227, -122.460213
+    37.764028, -122.510347
+
+### Author : [Lakhan Kumawat](https://github.com/Lakhankumawat)
```
Comment (kaustubhgupta) [inline Gmplot-Track the Route/README.md:35]:
```suggestion
## Author : [Lakhan Kumawat](https://github.com/Lakhankumawat)

```

File: Gmplot-Track the Route/main.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (kaustubhgupta) [inline Gmplot-Track the Route/main.py:1]: upload the CSV to be used in the script for testing

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

File: Gmplot-Track the Route/README.md (hunk 3 above)
Comment (Lakhankumawat) [reply Gmplot-Track the Route/README.md:35]: If i remove one # then size will be too big and it looks unusual .
If you say so i can remove .

File: Gmplot-Track the Route/main.py (hunk 4 above)
Comment (Lakhankumawat) [reply Gmplot-Track the Route/main.py:1]: I think my code has some bugs i will resolve them in a day or two and will notify you.

File: Gmplot-Track the Route/README.md (hunk 3 above)
Comment (kaustubhgupta) [reply Gmplot-Track the Route/README.md:35]: can remove this, we need to follow a consistent pattern

File: Gmplot-Track the Route/main.py (hunk 4 above)
Comment (kaustubhgupta) [reply Gmplot-Track the Route/main.py:1]: Okay, make sure to update it by tomorrow

Comment (Lakhankumawat) [reply Gmplot-Track the Route/main.py:1]: Actually i m out of town these days i will be back in 3 days so it would be great sir if you wait , please

Comment (kaustubhgupta) [reply Gmplot-Track the Route/main.py:1]: Oh okay, I will put the hold label for now.

**Comment by Lakhankumawat** [comment]
> I think now it looks fine , mentors please have a look
- Code Works Properly 
- CSV file added / Clarification
- Output Screenshot added in README

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]
You have not done something new here: https://pypi.org/project/gmplot/.  Only the coordinates loading is changes from zip to pandas data frame. This being level2 issue needs modifications. Suggestions:

- Add some more coordinates, Indian coordinates of around 100 places.
- Here: gmplot.GoogleMapPlotter(37.771260, -122.511011,17), take input for zoom level and for central coordinates, take the average of all the coordinates for x and y separately.

**Comment by Lakhankumawat** [comment]
> You have not done something new here: https://pypi.org/project/gmplot/. Only the coordinates loading is changed from zipping to pandas data frame. This being level2 issue needs modifications. Suggestions:
> 
> * Add some more coordinates, Indian coordinates of around 100 places.
> * Here: gmplot.GoogleMapPlotter(37.771260, -122.511011,17), take input for zoom level and for central coordinates, take the average of all the coordinates for x and y separately.

### Requested Changes are made 
- For now i haven't modified README screenshot , waiting for @kaustubhgupta sir to verify changes after that i will modify it.
- Code works fine 
- 100 Coordinates added
- Average Taken in x and y
- Zoom is taken as input <hr>
Screenshot (41)

**Comment by Lakhankumawat** [comment]
>Also i could have added polygon just like in the link you mentioned but it would have been lot more cluttering with 100 coordinates

**Comment by kaustubhgupta** [comment]
> > You have not done something new here: https://pypi.org/project/gmplot/. Only the coordinates loading is changed from zipping to pandas data frame. This being level2 issue needs modifications. Suggestions:
> > 
> > * Add some more coordinates, Indian coordinates of around 100 places.
> > * Here: gmplot.GoogleMapPlotter(37.771260, -122.511011,17), take input for zoom level and for central coordinates, take the average of all the coordinates for x and y separately.
> 
> ### Requested Changes are made
> * For now i haven't modified README screenshot , waiting for @kaustubhgupta sir to verify changes after that i will modify it.
> * Code works fine
> * 100 Coordinates added
> * Average Taken in x and y
> * Zoom is taken as input 
>   Screenshot (41)

This looks good. You can make the changes in the PR

**Comment by Lakhankumawat** [comment]
Updated README request a review from other members please if it is good to go.
For modifications let me know

File: Gmplot-Track the Route/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+2. plot the map 
+3. give output in html file format
+
+# Commands to use 
+
+| Command | README |
+| ------ | ------ |
+| `pip install gmplot` | [Gmplot/PyPI](https://pypi.org/project/gmplot/) |
+
+# CSV File Data
```
Comment (kaustubhgupta) [inline Gmplot-Track the Route/README.md:21]: Now you can remove this section and corresponding lat-long values

**Review by kaustubhgupta — APPROVED** [review]

**Comment by kaustubhgupta** [comment]
@Kushal997-das

File: Gmplot-Track the Route/README.md (hunk 5 above)
Comment (Lakhankumawat) [reply Gmplot-Track the Route/README.md]: Done ✅

**Review by antrikshmisri — APPROVED** [review]

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..

**Pull request merged**